src/peergos/shared/user/fs/FileWrapper.java

```diff
@@ -717,8 +717,11 @@ public static int getNumberOfChunks(long size) {
     }
 
     private CompletableFuture<Snapshot> recalculateThumbnail(Snapshot snapshot, Committer committer, String filename, AsyncReader fileData
-             , boolean isHidden, long fileSize, NetworkAccess network, WritableAbsoluteCapability fileWriteCap, Optional<byte[]> streamSecret
+             , boolean isHidden, long startIndex, long fileSize, NetworkAccess network, WritableAbsoluteCapability fileWriteCap, Optional<byte[]> streamSecret
     ) {
+        if(startIndex != 0) {
```

Comment:
So in this case you're not recalculating the thumbnail? If so, then why not just avoid calling this method at all?

Reply:
have refactored